### ISSUE TITLE
Find Method

### ISSUE THREAD

**Issue by TitanNano** [cocur/chain#14]
Please add a find method.

**Comment by gries** [comment]
Hi,
a search method is actually implemented but not documented.
`$chain->search('foo');`

**Comment by TitanNano** [comment]
I'm looking for something like 
```php
$chain->find(function ($item) use ($id) {
    return $item['id'] === $id;
});

**Comment by florianeckerstorfer** [comment]
Sounds like a good idea. Should we make `Cocur\Chain\Link\Search::search()` accept a callback or add a separate function `find()`?

Opinions?

**Comment by TitanNano** [comment]
to have one method is probably less confusing. So i'd say extend `search()`

**Comment by gries** [comment]
hm changing the search method would either require a BC break or a very wonky signature:
`search($needle, $strict = false, callable $matcher)`

**Comment by florianeckerstorfer** [comment]
I don't think this is a BC break. The signature of `search` is `mixed`, we check first if `$needle` is a callback and `array_search` in the else.

**Comment by gries** [comment]
ok sounds good, maybe we can utilize http://php.net/manual/en/function.array-filter.php for the implementation as it nearly covers the functionality we're looking for.

**Comment by gries** [comment]
i've written an implementation but I'm still not sure if `search` is the right way to go for,
the current behavior of search is that it returns the matching *key* of the item inside the array.

So:
```
$chain = new Chain['foo', 'bar'];
$chain->search('bar'); // returns 1
```
My current implementation behaves the same way:
```
$chain = new Chain['foo', 'bar'];
$chain->search(function($item) {
  return $item === 'bar';
}); // returns 1
```
I think what @TitanNano want's is to return *'bar'*

**Comment by TitanNano** [comment]
@gries yes that's what I'm looking for.

**Comment by florianeckerstorfer** [comment]
Good point @gries . Then we should add a new method `find()`.

**Comment by florianeckerstorfer** [comment]
Fixed in #17, thanks @gries
